zcutil/build.sh

```diff
@@ -2,19 +2,35 @@
 
 set -eu -o pipefail
 
+function cmd_pref() {
+    if type -p "$2" > /dev/null; then
+        eval "$1=$2"
+    else
+        eval "$1=$3"
+    fi
+}
+
+# If a g-prefixed version of the command exists, use it preferentially.
+function gprefix() {
+    cmd_pref "$1" "g$2" "$2"
+}
+
+gprefix READLINK readlink
+cd "$(dirname "$("$READLINK" -f "$0")")/.."
+
 # Allow user overrides to $MAKE. Typical usage for users who need it:
-#   MAKE=gmake ./zcutil/build.sh -j$(nproc)
+#   MAKE=custom-make ./zcutil/build.sh -j$(nproc)
 if [[ -z "${MAKE-}" ]]; then
-    MAKE=make
+    gprefix MAKE make
 fi
 
 # Allow overrides to $BUILD and $HOST for porters. Most users will not need it.
 #   BUILD=i686-pc-linux-gnu ./zcutil/build.sh
 if [[ -z "${BUILD-}" ]]; then
```

Comment:
what does this syntax mean in bash?

Reply:
It causes the brackets to expand to the empty string if `BUILD` is unset. See https://stackoverflow.com/a/32964575

Reply:
and empty string is considered true for `if`?

Reply:
ok the `-z` turns an empty string to true.